### ISSUE TITLE
False positives for unused variables

### ISSUE THREAD

**Issue by Goldziher** [snok/flake8-type-checking#116]
Hiya,

So I had an unused variable, which I had to cleanup. But this is picked by this plugin as a typing variable:

```shell
starlite/signature.py:22:1: F401 'starlite.constants.EXTRA_KEY_IS_READ_ONLY' imported but unused
starlite/signature.py:22:1: TC001 Move application import 'starlite.constants.EXTRA_KEY_IS_READ_ONLY' into a type-checking block
```

**Comment by sondrelg** [comment]
Yep, I'll agree this is a little bit annoying. Could try to add handling for this :+1:

**Comment by Goldziher** [comment]
The point is that this plugin should not identify unused imports as used for typing, unless they are used for typing - in which case they wouldnt be unused.

**Comment by sondrelg** [comment]
v2.1.2 should be out with a fix shortly 🙂
